### PR TITLE
FIX-8: Install nginx, added nginx.conf file serves static files and proxy pass then updated entrypoint script and collecting static files before running entrypoint script.

### LINKED ISSUE #8: Add nginx
Currently, we serve our application directly by exposing the gunicorn from our container. Due to this, we are not able to serve our static file as Django doesn't support and recommend serving static files in production on its own. We need to use services like apache or nginx.

- [ ] Install Nginx inside the container
- [ ] Create `nginx.conf`
    - [ ] Serve static file under tha url written in `settings.py`
    - [ ] proxy pass rest of the request to the gunicorn running on port 8000
- [ ] update entrypoint script
    - [ ] enable our nginx conf
    - [ ] run collect static before spinning up the server

### PR DESCRIPTION
**Pull request by A4ANK**
Signed-off-by: ANKUR DHAKAR <dhakar.ankur18@gmail.com>

Fixes #8 

Changes:   Install nginx, added nginx.conf file serves static files and proxy pass also updated root path for collecting static in settings then updated entrypoint script and collecting static files before running entrypoint script.

### CODE REVIEW

File: poker/settings.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['0.0.0.0','192.168.56.27']
```
Comment (jatinkatyal13) [inline poker/settings.py]: Please add * as of now rather than hardcoding IPs

File: nginx.conf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    location /static/ {
+        root /opt/poker/static/;
+  }
+}
+
+server {
+  listen 80;
+  server_name 0.0.0.0:8000;
+
```
Comment (jatinkatyal13) [inline nginx.conf:4]: add the `/static` location here itself and use alias directive rather than root

File: nginx.conf [hunk 3]
```diff
@@ -0,0 +1,22 @@
+upstream poker  {
```
Comment (jatinkatyal13) [inline nginx.conf]: remove unused upstream declaration

File: docker-compose.yml [hunk 4]
```diff
@@ -2,15 +2,15 @@ version : '3.2'
 
 services:
   postgres:
-    image: 'postgres:9.6'
+    image: 'postgres:10'
```
Comment (jatinkatyal13) [inline docker-compose.yml]: Let's stick with current compose file

File: Dockerfile [hunk 5]
```diff
@@ -12,5 +12,7 @@ COPY Pipfile.lock Pipfile.lock
 RUN pipenv install --system
 
 COPY . .
+RUN mkdir -p /run/nginx
```
Comment (jatinkatyal13) [inline Dockerfile]: Why do we need this folder?

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]

File: start.sh [hunk 6]
```diff
@@ -2,4 +2,6 @@
 
 /usr/sbin/crond
 python manage.py migrate
-gunicorn poker.wsgi -b 0.0.0.0:8000
\ No newline at end of file
+yes yes| python manage.py collectstatic
```
Comment (jatinkatyal13) [inline start.sh]: remove `yes yes`

File: docker-compose.yml [hunk 7]
```diff
@@ -13,5 +13,5 @@ services:
     image: 'codingblocks/poker'
     ports:
       - "8000:8000"
-    env_file: 
-      - '.env'
+#    env_file:
```
Comment (jatinkatyal13) [inline docker-compose.yml]: remove comments it is a required config

File: Dockerfile (hunk 5 above)
Comment (jatinkatyal13) [inline Dockerfile]: we don't need to create this directory

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]

**Comment by A4ANK** [comment]
Please review this pull request.

**Pull request closed without merging**